src/providers/core/mod.rs

```diff
@@ -99,7 +104,12 @@ impl Provide for Provider {
 
         let mut clients: Vec<String> = Vec::new();
         for provider in &self.prov_list {
-            let mut result = provider.list_clients(_op)?;
+            let mut result = provider.list_clients(_op).unwrap_or_else(|e| {
+                error!("list_clients failed on provider with {}", e);
```

Comment:
Any way to include the provider ID in that error message?

Reply:
(same for the other methods that you changed)

Reply:
I could do it with the `describe` method? I tried that rapidly but felt like it was going too far for this purpose

Reply:
Yeah, was wondering about that... I'm just thinking that since the operation is unlikely to proper fail and since logs are the only thing that tells you what fails maybe having more specific messages would help